plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/DeleteResourceHandler.java

```diff
@@ -48,16 +49,24 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 			return OpenShiftUIActivator.statusFactory().cancelStatus("No resource selected that we can delete."); //$NON-NLS-1$
 		}
 		try (Stream<IResourceUIModel> stream = Arrays.stream(resources)) {
-	        boolean hasProject = stream.anyMatch(resource -> resource instanceof IProjectAdapter);
+	        boolean willDeleteSubResources = stream.anyMatch(resource -> Platform.getAdapterManager().getAdapter(resource.getResource(), ResourceProcessor.class).willCascadeDeleteLinkedResources());
```

Comment:
Sub resources for a Deployment/Service are not found:

![](http://content.screencast.com/users/fbricon/folders/Jing/media/7651f75c-f308-426c-b021-c4a429e60a35/00000438.png)